src/main/java/org/opensearch/agent/tools/LogPatternTool.java

```diff
@@ -0,0 +1,294 @@
+     * @param rawString string value of the field to generate a pattern
+     * @param pattern @Nullable the specified pattern to remove, use DEFAULT_IGNORED_CHARS if null
+     * @return the generated pattern value
+     */
+    @VisibleForTesting
+    static String extractPattern(String rawString, Pattern pattern) {
+        if (pattern != null)
+            return pattern.matcher(rawString).replaceAll("");
+        char[] chars = rawString.toCharArray();
+        int pos = 0;
+        for (int i = 0; i < chars.length; i++) {
+            if (!DEFAULT_IGNORED_CHARS.contains(chars[i])) {
+                chars[pos++] = chars[i];
+            }
+        }
+        return new String(chars, 0, pos);
+    }
+
+    /**
+     * Find the longest field in the sample log source. This function imitates the same logic of
+     * Observability log pattern feature here:
+     * <a href="https://github.com/opensearch-project/dashboards-observability/blob/279463ad0c8e00780bf57f85e6d2bf30e8c7c93d/public/components/event_analytics/hooks/use_fetch_patterns.ts#L137C2-L164C6">setDefaultPatternsField</a>
+     * @param sampleLogSource sample log source map
+     * @return the longest field name
+     */
+    @VisibleForTesting
+    static String findLongestField(Map<String, Object> sampleLogSource) {
+        String longestField = null;
+        int maxLength = 0;
+
+        for (Map.Entry<String, Object> entry : sampleLogSource.entrySet()) {
+            Object value = entry.getValue();
+            if (value instanceof String) {
+                String stringValue = (String) value;
+                int length = stringValue.length();
+                if (length > maxLength) {
+                    maxLength = length;
+                    longestField = entry.getKey();
+                }
+            }
+        }
+        return longestField;
+    }
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    @Override
+    public boolean validate(Map<String, String> parameters) {
+        return super.validate(parameters) && parameters.containsKey(INDEX_FIELD) && !StringUtils.isBlank(parameters.get(INDEX_FIELD));
+    }
+
+    private static int getPositiveInteger(Map<String, ?> params, String paramName) {
+        int value = getInteger(params, paramName);
+        checkPositive(value, paramName);
+        return value;
+    }
+
+    private static int getInteger(Map<String, ?> params, String paramName) {
+        int value;
+        try {
+            value = Integer.parseInt((String) params.get(paramName));
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException(
+                LoggerMessageFormat.format("Invalid value {} for parameter {}, it should be a number", params.get(paramName), paramName)
+            );
+        }
+        return value;
+    }
+
+    private static void checkPositive(int value, String paramName) {
+        if (value < 0) {
```

Comment:
value <= 0

Reply:
It make < 0 on purpose as I found many places in OpenSearch using "<0" for checking positive. See some examples below:

1. https://github.com/opensearch-project/OpenSearch/blob/e8b02c9545d7908ae1a3ea2d109c06ff3e0c07a5/server/src/main/java/org/opensearch/index/reindex/BulkByScrollTask.java#L917

2. org.opensearch.jobscheduler.repackage.com.cronutils.utils.Preconditions::checkPositive
```
  public static void checkPositive(int value, Object errorMessage) {
    if (value < 0) {
      throw new IllegalArgumentException(String.valueOf(errorMessage));
    }
  }
```

Reply:
Then rename it to `checkNonNegative`

Reply:
I wonder if it's a habit to check '< 0' with function name `checkPostive`.

Reply:
It's consistent to check `<0` in method `checkPositive`, but in our case, 0 doesn't make sense either for top_n or sample_count, right? And in this case, we should let user know their configuration issue.